aya-obj/src/btf/btf.rs

```diff
@@ -425,6 +425,18 @@ impl Btf {
                     fixed_ty.name_offset = 0;
                     types.types[i] = BtfType::Ptr(fixed_ty)
                 }
+                // Fixup STRUCT for Rust
+                // LLVM emits names that include characters that aren't valid C idents.
+                // Rather than replace all unsupported characters, just omit the name entirely.
+                BtfType::Struct(s) => {
+                    let name = self.string_at(s.name_offset)?.to_string();
+                    if name.contains(['<', '>', ' ', ':', ',']) {
```

Comment:
Shouldn't this be applied to everything that uses generics? functions, enums, etc not only structs

Reply:
Good question, I personally had only run into this with Struct's but I guess it could apply to the following (maybe I'm missing some) 

```
#define BTF_KIND_PTR            2       /* Pointer      */
#define BTF_KIND_STRUCT         4       /* Struct       */
#define BTF_KIND_UNION          5       /* Union        */
#define BTF_KIND_ENUM           6       /* Enumeration up to 32-bit values */
#define BTF_KIND_FUNC           12      /* Function     */
#define BTF_KIND_FUNC_PROTO     13      /* Function Proto       */
#define BTF_KIND_VAR            14      /* Variable     */
#define BTF_KIND_ENUM64         19      /* Enumeration up to 64-bit values */
```

Reply:
The way I would go about this is:

- create a rust program that uses all items that can include generics (presumably you already have a program that hits the Struct case, so you could extend that)
- generate BTF for it
- have an integration test that loads the program

This should give you all the possible errors until you fix all of them 😅

Reply:
Will do!